pkg/apis/application/v1alpha1/types.go

```diff
@@ -2507,3 +2509,34 @@ func (d *ApplicationDestination) MarshalJSON() ([]byte, error) {
 	}
 	return json.Marshal(&struct{ *Alias }{Alias: (*Alias)(dest)})
 }
+
+// InstanceName returns the name of the application that can be used as value
+// for a tracking label
+func (a *Application) InstanceName() string {
+	// When app has no namespace set, or the namespace is the default ns, we
```

Comment:
"or the namespace is the default ns" - that part doesn't seem true here. Does `""` imply "default ns"?

Reply:
I think this method is deprecated and fully replaced by `InstanceNameWithDefault()`. It's a left-over from the previous PR's code which I used as a base.

Probably it makes sense to rename `InstanceNameWithDefault()` to `InstanceName()`, let me check if VS code makes this refactor easy :)

Reply:
Done.